File: app/src/main/java/com/apps/kunalfarmah/furnituremagic/ViewModels/FurnitureViewModel.java
package com.apps.kunalfarmah.furnituremagic.ViewModels;

import android.app.Application;

import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;


import com.apps.kunalfarmah.furnituremagic.Repositories.FurnitureRepository;
import com.apps.kunalfarmah.furnituremagic.Room.Furniture;

import org.jetbrains.annotations.NotNull;

import java.util.List;

public class FurnitureViewModel extends AndroidViewModel {

    private FurnitureRepository postRoomDBFurnituresitory;
    private LiveData<List<Furniture>> mAllFurnitures;

    public FurnitureViewModel(Application application) {
        super(application);
        postRoomDBFurnituresitory = new FurnitureRepository(application);
    }

    public LiveData<List<Furniture>> getFurnitures() {
        mAllFurnitures = postRoomDBFurnituresitory.getFurnitures();
        return mAllFurnitures;
    }

    public void insertItem(@NotNull Furniture furniture) {
        postRoomDBFurnituresitory.insert(furniture);
    }
}

File: app/src/main/java/com/apps/kunalfarmah/furnituremagic/ViewModels/FurnitureViewModel.kt
package com.apps.kunalfarmah.furnituremagic.ViewModels

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import com.apps.kunalfarmah.furnituremagic.Repositories.FurnitureRepository
import com.apps.kunalfarmah.furnituremagic.Room.Furniture

class FurnitureViewModel(application: Application) : AndroidViewModel(application) {
    private val postRoomDBFurnituresitory: FurnitureRepository
    private var mAllFurnitures: LiveData<List<Furniture?>?>? = null
    val furnitures: LiveData<List<Furniture?>?>
        get() {
            mAllFurnitures = postRoomDBFurnituresitory.furnitures
            return mAllFurnitures!!
        }

    fun insertItem(furniture: Furniture) {
        postRoomDBFurnituresitory.insert(furniture)
    }

    fun deleteItem(furniture: Furniture?) {
        postRoomDBFurnituresitory.delete(furniture)
    }

    fun deleteALL() {
        postRoomDBFurnituresitory.deleteAll()
    }

    init {
        postRoomDBFurnituresitory = FurnitureRepository(application)
    }
}